src/repl.rs

```diff
@@ -367,162 +368,157 @@ impl<F: LurkField> ReplTrait<F> for ReplState<F> {
+                            Some(new_name)
+                        }
+                        "DEFREC" => {
+                            // Extends env with a recursive binding.
+                            //
+                            // This: !(:def foo (lambda () 123))
+                            //
+                            // Gets macroexpanded to this: (letrec ((foo (lambda () 123)))
+                            //                               (current-env))
+                            //
+                            // And the state's env is set to the result.
+                            let (first, rest) = store.car_cdr(&rest)?;
+                            let (second, rest) = store.car_cdr(&rest)?;
+                            assert!(rest.is_nil());
+                            let l = store.sym("LETREC");
+                            let current_env = store.sym("CURRENT-ENV");
+                            let binding = store.list(&[first, second]);
+                            let bindings = store.list(&[binding]);
+                            let current_env_call = store.list(&[current_env]);
+                            let expanded = store.list(&[l, bindings, current_env_call]);
+                            let (expanded_evaled, _, _, _) = self.eval_expr(expanded, store)?;
+
+                            self.env = expanded_evaled;
+
+                            let (new_binding_outer, _) = store.car_cdr(&expanded_evaled)?;
+                            let (new_binding_inner, _) = store.car_cdr(&new_binding_outer)?;
+                            let (new_name, _) = store.car_cdr(&new_binding_inner)?;
+                            Some(new_name)
+                        }
+                        "LOAD" => {
+                            match store.fetch(&store.car(&rest)?).unwrap() {
+                                Expression::Str(path) => {
+                                    let joined = p.as_ref().join(Path::new(&path));
+                                    self.handle_load(store, &joined, package)?
                                 }
-                                io::stdout().flush().unwrap();
-                                None
-                            }
-                            "SET-ENV" => {
-                                // The state's env is set to the result of evaluating the first argument.
-                                let (first, rest) = store.car_cdr(&rest)?;
-                                assert!(rest.is_nil());
-                                let (first_evaled, _, _, _) = self.eval_expr(first, store)?;
-                                self.env = first_evaled;
-                                None
-                            }
-                            _ => {
-                                panic!("!({} ...) is unsupported.", s.name());
+                                _ => bail!("Argument to :LOAD must be a string."),
```

Comment:
Nit
```suggestion
                                _ => bail!("Argument to LOAD must be a string."),
```

Reply:
Whoops, indeed. This automerged already, but will fix.